frontend/src/components/pages/search/hooks/useTrekResults.ts

```diff
@@ -27,14 +27,14 @@ const computeUrl = (
   textFilter: string | null,
   dateFilter: DateFilter | null,
 ) => {
-  const urlParams =
-    textFilter !== null
-      ? [...formatFiltersUrl(filtersState), `text=${textFilter}`]
-      : formatFiltersUrl(filtersState);
+  const urlParams = formatFiltersUrl(filtersState);
 
-  dateFilter && dateFilter.beginDate !== '' && urlParams.push(`beginDate=${dateFilter?.beginDate}`);
-  dateFilter && dateFilter.endDate !== '' && urlParams.push(`endDate=${dateFilter?.endDate}`);
-  const formattedUrl = `search?${urlParams.join('&')}`;
+  textFilter && urlParams.push(`text=${textFilter}`);
+  dateFilter?.beginDate && urlParams.push(`beginDate=${dateFilter.beginDate}`);
+  dateFilter?.endDate && urlParams.push(`endDate=${dateFilter.endDate}`);
+
+  const params = urlParams.join('&');
+  const formattedUrl = `search${params && '?'}${params}`;
```

Comment:
If params is truthy → "search?PARAMSTRING"
If params is falsy (so only empty string because of `join('&')`) → "searchEMPTYSTRING"


So:
```suggestion
  const formattedUrl = params ? `search?${params}` : 'search';
```